WcaOnRails/app/models/competition.rb

```diff
@@ -305,6 +305,20 @@ def events
     eventSpecs.split.map { |e| Event.find_by_id(e.split("=")[0]) }.sort_by &:rank
   end
 
+  def has_event?(event)
+    event.id.in?(self.events.map(&:id))
+  end
+
+  def belongs_to_region?(region_or_country)
+    (self.countryId == region_or_country) || (self.country.continentId == region_or_country)
+  end
+
+  def search(search_param)
+    if search_param != nil
```

Comment:
Why are you checking if search_param is not null here? When does this get called with null? What does `search` mean? I'd expect a search method to reutrn an array of results or something, but this looks like it just returns true or false....